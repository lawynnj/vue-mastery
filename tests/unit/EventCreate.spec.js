import Vuex from 'vuex'
import EventCreate from '@/views/EventCreate.vue'

import { shallowMount, createLocalVue } from '@vue/test-utils'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('EventCreate', () => {
  let actions, store
  beforeEach(() => {
    actions = {
      createEvent: jest.fn()
    }

    const event = {
      namespaced: true,
      actions
    }

    store = new Vuex.Store({
      modules: {
        event
      },
      state: {
        categories: []
      }
    })
  })

  test('createEvent() is invoked on submit', () => {
    const createEventMock = jest.fn()
    const wrapper = shallowMount(EventCreate, {
      attachToDocument: true,
      store,
      localVue,
      methods: {
        createEvent: createEventMock
      }
    })
    wrapper.find('form').trigger('submit.prevent')
    expect(createEventMock).toHaveBeenCalled()
  })

  test('EventCreate renders correctly', () => {
    // const createFreshEventObjectMock = jest.fn()
    // createFreshEventObjectMock.mockReturnValue(() => ({}))

    const wrapper = shallowMount(EventCreate, {
      store,
      localVue,
      methods: {
        // createFreshEventObject: createFreshEventObjectMock
      }
    })
    wrapper.setData({
      time: [],
      event: {}
    })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
