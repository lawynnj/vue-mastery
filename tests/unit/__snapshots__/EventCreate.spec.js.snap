// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCreate EventCreate renders correctly 1`] = `
<div
  class="event-create"
>
  <h1>
    Create Component
  </h1>
   
  <form>
    <label>
      Select a category
    </label>
     
    <select />
     
    <h3>
      Name & describe your event
    </h3>
     
    <div
      class="field"
    >
      <label>
        Title
      </label>
       
      <input
        placeholder="Add an event title"
        type="text"
      />
    </div>
     
    <div
      class="field"
    >
      <label>
        Description
      </label>
       
      <input
        placeholder="Add a description"
        type="text"
      />
    </div>
     
    <h3>
      Where is your event?
    </h3>
     
    <div
      class="field"
    >
      <label>
        Location
      </label>
       
      <input
        placeholder="Add a location"
        type="text"
      />
    </div>
     
    <h3>
      When is your event?
    </h3>
     
    <div
      class="field"
    >
      <label>
        Date
      </label>
       
      <datepicker-stub
        format="dd MMM yyyy"
        language="[object Object]"
        maximumview="year"
        minimumview="day"
        placeholder="Select a date"
        value=""
      />
    </div>
     
    <div
      class="field"
    >
      <label>
        Select a time
      </label>
       
      <select>
        <option>
          0:00
        </option>
        <option>
          1:00
        </option>
        <option>
          2:00
        </option>
        <option>
          3:00
        </option>
        <option>
          4:00
        </option>
        <option>
          5:00
        </option>
        <option>
          6:00
        </option>
        <option>
          7:00
        </option>
        <option>
          8:00
        </option>
        <option>
          9:00
        </option>
        <option>
          10:00
        </option>
        <option>
          11:00
        </option>
        <option>
          12:00
        </option>
        <option>
          13:00
        </option>
        <option>
          14:00
        </option>
        <option>
          15:00
        </option>
        <option>
          16:00
        </option>
        <option>
          17:00
        </option>
        <option>
          18:00
        </option>
        <option>
          19:00
        </option>
        <option>
          20:00
        </option>
        <option>
          21:00
        </option>
        <option>
          22:00
        </option>
        <option>
          23:00
        </option>
      </select>
    </div>
     
    <input
      class="button -fill-gradient"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
